internal/observability/event/event_type.go

```diff
@@ -24,3 +27,11 @@ func (et EventType) Validate() error {
 		return fmt.Errorf("%s: '%s' is not a valid event type: %w", op, et, ErrInvalidParameter)
 	}
 }
+
+// GenerateNodeID generates a new UUID.
```

Comment:
Might be worth describing how an `eventlogger.NodeID` compares to a regular ol' UUID.

Reply:
It doesn't by much. The only value is that this function casts the string returned by uuid.GenerateUUID to the eventlogger.NodeID type (which is a type derived from string).